ql/src/test/results/clientpositive/perf/tez/query61.q.out

```diff
@@ -165,70 +167,81 @@ Stage-0
                                     <-Map 17 [SIMPLE_EDGE] vectorized
-                                      SHUFFLE [RS_278]
+                                      PARTITION_ONLY_SHUFFLE [RS_272]
                                         PartitionCols:_col0
-                                        Select Operator [SEL_277] (rows=50 width=4)
+                                        Select Operator [SEL_271] (rows=50 width=4)
                                           Output:["_col0"]
-                                          Filter Operator [FIL_276] (rows=50 width=12)
+                                          Filter Operator [FIL_270] (rows=50 width=12)
                                             predicate:((d_year = 1999) and (d_moy = 11) and d_date_sk is not null)
                                             TableScan [TS_9] (rows=73049 width=12)
                                               default@date_dim,date_dim,Tbl:COMPLETE,Col:COMPLETE,Output:["d_date_sk","d_year","d_moy"]
                                     <-Map 9 [SIMPLE_EDGE] vectorized
-                                      SHUFFLE [RS_274]
+                                      SHUFFLE [RS_283]
                                         PartitionCols:_col0
-                                        Select Operator [SEL_272] (rows=479120969 width=126)
+                                        Select Operator [SEL_281] (rows=479120969 width=126)
                                           Output:["_col0","_col1","_col2","_col3","_col4","_col5"]
-                                          Filter Operator [FIL_270] (rows=479120969 width=126)
-                                            predicate:(ss_sold_date_sk is not null and ss_promo_sk is not null and ss_customer_sk is not null and ss_store_sk is not null and ss_item_sk is not null)
+                                          Filter Operator [FIL_279] (rows=479120969 width=126)
+                                            predicate:(ss_sold_date_sk is not null and ss_promo_sk is not null and ss_customer_sk is not null and ss_store_sk is not null and ss_item_sk is not null and ss_sold_date_sk BETWEEN DynamicValue(RS_22_date_dim_d_date_sk_min) AND DynamicValue(RS_22_date_dim_d_date_sk_max) and in_bloom_filter(ss_sold_date_sk, DynamicValue(RS_22_date_dim_d_date_sk_bloom_filter)) and ((ss_sold_date_sk is not null and ss_promo_sk is not null and ss_customer_sk is not null and ss_store_sk is not null and ss_item_sk is not null) or (ss_sold_date_sk is not null and ss_customer_sk is not null and ss_store_sk is not null and ss_item_sk is not null)))
                                             TableScan [TS_6] (rows=575995635 width=126)
                                               default@store_sales,store_sales,Tbl:COMPLETE,Col:COMPLETE,Output:["ss_sold_date_sk","ss_item_sk","ss_customer_sk","ss_store_sk","ss_promo_sk","ss_ext_sales_price"]
+                                            <-Reducer 18 [BROADCAST_EDGE] vectorized
```

Comment:
Maybe it was because one TS was targeted by a SJ and the other one was not?

Reply:
the interesting thing is that this plan has no scans without SJ -  I think during one of the merge steps the SJ filter was removed - and because older TS ops were not removed in a later step it merged with it...

but this patch fixes this bug for sure...so we are good :)